src/Nethermind/Nethermind.Core/Tasks/WaitForPassingTasks.cs

```diff
@@ -0,0 +1,48 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Nethermind.Core.Tasks;
+
+public static class Wait
+{
+    /// <summary>
+    /// Wait for any of the task that passed the predicate and forward the result, or all of the task to complete.
+    /// </summary>
+    /// <param name="cond"></param>
+    /// <param name="tasks"></param>
+    /// <typeparam name="T"></typeparam>
+    /// <returns></returns>
+    public static async Task<T> ForPassingTask<T>(Func<T, bool> cond, params IEnumerable<Task<T>> tasks)
```

Comment:
Can we make it span?
LINQ style naming? `AnyWhere` or just `Any` - in LINQ it has optional parameter:https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.any?view=net-8.0#system-linq-enumerable-any-1(system-collections-generic-ienumerable((-0))-system-func((-0-system-boolean)))
```suggestion
    public static async Task<T> AnyWhere<T>(Func<T, bool> predicate, params ReadOnlySpan<Task<T>> tasks)
```

Reply:
Because its an async code.

Reply:
Ok the naming comment still stands